src/storage/src/client.rs

```diff
@@ -0,0 +1,297 @@
+///   Note that object names can contain `/` characters, which are treated as
+///   any other character (no special directory semantics).
+///
+/// [with_endpoint()]: super::builder::storage::ClientBuilder::with_endpoint
+/// [with_credentials()]: super::builder::storage::ClientBuilder::with_credentials
+/// [Private Google Access with VPC Service Controls]: https://cloud.google.com/vpc-service-controls/docs/private-connectivity
+/// [Application Default Credentials]: https://cloud.google.com/docs/authentication#adc
+#[derive(Clone, Debug)]
+pub struct Storage {
+    inner: super::generated::gapic::client::Storage,
+}
+
+impl Storage {
+    /// Returns a builder for [Storage].
+    ///
+    /// ```no_run
+    /// # tokio_test::block_on(async {
+    /// # use google_cloud_storage::client::Storage;
+    /// let client = Storage::builder().build().await?;
+    /// # gax::Result::<()>::Ok(()) });
+    /// ```
+    pub fn builder() -> ClientBuilder {
+        gax::client_builder::internal::new_builder(client_builder::Factory)
+    }
+
+    /// Permanently deletes an empty bucket.
+    ///
+    /// This request will fail if the bucket is not empty. You must manually
+    /// delete the objects in a bucket (including archived and soft deleted
+    /// objects) before deleting the bucket.
+    ///
+    /// # Parameters
+    /// * `name` - the bucket name. In `projects/_/buckets/{bucket_id}` format.
+    ///
+    /// # Example
+    /// ```
+    /// # use google_cloud_storage::client::Storage;
+    /// async fn example(client: &Storage) -> gax::Result<()> {
+    ///     client.delete_bucket("projects/_/buckets/my-bucket").send().await?;
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn delete_bucket<T: Into<String>>(&self, name: T) -> super::builder::storage::DeleteBucket {
+        self.inner.delete_bucket().set_name(name)
+    }
+
+    /// Returns metadata for the specified bucket.
+    ///
+    /// # Parameters
+    /// * `name` - the bucket name. In `projects/_/buckets/{bucket_id}` format.
+    ///
+    /// # Example
+    /// ```
+    /// # use google_cloud_storage::client::Storage;
+    /// async fn example(client: &Storage) -> gax::Result<()> {
+    ///     let bucket = client.get_bucket("projects/_/buckets/my-bucket").send().await?;
+    ///     assert_eq!(&bucket.name, "projects/_/buckets/my-bucket");
+    ///     println!("bucket details={bucket:?}");
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn get_bucket<T: Into<String>>(&self, name: T) -> super::builder::storage::GetBucket {
+        self.inner.get_bucket().set_name(name)
+    }
+
+    /// Creates a new bucket.
+    ///
+    /// # Parameters
+    /// * `name` - the bucket name. In `projects/_/buckets/{bucket_id}` format.
+    ///
+    /// # Example
+    /// ```
+    /// # use google_cloud_storage::client::Storage;
+    /// async fn example(client: &Storage) -> gax::Result<()> {
+    ///     let bucket = client.create_bucket("projects/my-project", "my-bucket").send().await?;
```

Comment:
Are we concerned about handwritten sample code that doesn't run?

We could maybe use `#[doc(include = example.rs)` for the sample body, and run the examples in our integration tests.

Also, this is a thing: https://doc.rust-lang.org/stable/rustdoc/scraped-examples.html

(Non-blocking for this PR, but potentially something we want in the long term)